housekeeping/src/main/java/com/hotels/housekeeping/model/EntityLegacyReplicaPath.java

```diff
@@ -46,6 +46,12 @@
   @Column(name = "path_event_id", nullable = true, length = 250)
   protected String pathEventId;
 
+  @Column(name = "metastore_database_name")
```

Comment:
Shouldn't these both be `nullable = true`? For this to work if you upgrade and you have an existing table surely it adds two optional columns to the table with null as the default value?

Reply:
Or is `nullable = true` the default? In which case we could remove it from `pathEventId`.

Reply:
What type do you get for these fields if you don't specify a length? I'm fine with going with this it's just making me wonder why we put limits on things like the full path up above (and if we're going to pay for this later). I've got a feeling we did some premature optimisations on the above and could end up paying for it later (separate issue to what you're doing here, it's just made me think about it).

Reply:
default is 255 length and true

Reply:
OK, so the nullable is fine then. I'd suggest either adding it here so that we explicitly document it on all members, or remove the `nullable = true` above and just leave the `nullable = false` (my vote goes for the latter - convention over configuration).

So how do we feel about the 255 limit? I guess it's OK, looking at the Hive internal schema for 2.1 (grep for DB_NAME, TABLE_NAME, TBL_NAME) it seems to be varchar(128).

Reply:
Hmm, in Hive 2.3 the values go up to 256 - https://github.com/apache/hive/blob/master/metastore/scripts/upgrade/mysql/hive-schema-2.3.0.mysql.sql#L586

Reply:
And then for 3.0.0 and above I can only find "Hive" versions of the DDL which mark the values as "string" (e.g. https://github.com/apache/hive/blob/master/metastore/scripts/upgrade/hive/hive-schema-3.0.0.hive.sql and not any db-specific ones). Anyway, I think we should either increase them to something like 512 (which should be more than enough, famous last words and all) or do what it takes to make them "TEXT" fields of unlimited length in the underlying DB (although it will be a bit weird that these are TEXT while path which will almost certainly always be longer is VARCHAR with a limit).

Reply:
it's called metastore potentially being not tied to hive for a reason. But yeah 512 is probably ok.